src/common/security/robot/robot.go

```diff
@@ -40,8 +41,12 @@ func filterPolicies(namespace types.Namespace, policies []*types.Policy) []*type
 	for _, policy := range policies {
 		if types.ResourceAllowedInNamespace(policy.Resource, namespace) {
 			results = append(results, policy)
+			// give the PUSH action a pull access
+			if policy.Action == rbac.ActionPush {
+				pull := &types.Policy{Resource: namespace.Resource(rbac.ResourceRepository), Action: rbac.ActionPull}
```

Comment:
Copy from the policy and modify its action to `rbac.ActionPull`

Reply:
gotcha, updated.